src/test/functional/functional-tests.js

```diff
@@ -65,6 +65,9 @@ describe('functional tests', function() {
   // a directory with files to read from, i.e. /mint/data.
   var dataDir = process.env['MINT_DATA_DIR']
 
+  // default partSize is 64Mb, use 5Mb here to test multipart upload
+  playConfig.partSize = 5 * 1024 * 1024
```

Comment:
!789 changes big test file size from 6Mb to 65Mb. But putObject still do single upload due to undefined content-length.

Here I force partSize to ensure upload is done with multipart. So I propose (in another PR if accepted) to reduce the 65Mb test file to optimize those test. We could return to 6Mb file.

Reply:
> #789 changed test file size from 6Mb to 65Mb. But putObject still does single upload due to undefined content-length.

Hi @sberthier, my apologies for the late review, what was the problem you were seeing with the undefined content-length? As you commented, `partSize` should normally default to 64MB based on src/main/minio.js:132

Reply:
I see multipart upload happening fine by leaving the partSize to default.

Reply:
> what was the problem you were seeing with the undefined content-length?

In `putObject`, an optional `size` argument allow chunk size computation in case of stream. For buffer or string, length is implicit. If this size is undefined, `maxObjectSize` is taken by default, which is 5To. It results in chunk size of about 500Mo. 

Setting the `partSize` force it to be that one (`overRidePartSize = true` is setted in constructor and used in `calculatePartSize`)

There is not the same problem for `fPutObject`

Reducing `partSize` here to 5Mb is juste to optimize test. I can postpone it in another PR. I can set it to 64Mb still.

Reply:
@sberthier could you please send the partSize related changes as separate PR. So that we can take this PR in.

Reply:
@kanagarajkm I set the partSize to default (64Mb) because it is necessary for my new test.